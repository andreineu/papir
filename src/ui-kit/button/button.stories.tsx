import type { Meta, StoryObj } from '@storybook/react';

import type { ButtonProps } from './button';
import { Button } from './button';

const meta: Meta<typeof Button> = {
  component: Button,
};

export default meta;
type Story = StoryObj<typeof Button>;

export const Primary: Story = {
  args: {
    children: 'Primary',
  },
};

const variants: ButtonProps['variant'][] = [
  'default',
  'destructive',
  'ghost',
  'link',
  'outline',
  'secondary',
];

const sizes: ButtonProps['size'][] = ['sm', 'default', 'lg'];

export const AllVariants: Story = {
  render: () => (
    <>
      <div className="flex gap-2">
        {variants.map((variant) => (
          <div key={variant} className="flex flex-col items-center gap-2">
            {sizes.map((size) => (
              <Button key={size} variant={variant} size={size}>
                {variant}
              </Button>
            ))}
          </div>
        ))}
      </div>
    </>
  ),
};
