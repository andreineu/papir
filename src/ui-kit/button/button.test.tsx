import React from 'react';

import { fireEvent, render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';

import { Button } from './button';

describe('ui-button', () => {
  it('renders without crashing', () => {
    const { getByTestId } = render(<Button />);

    const buttonElement = getByTestId('button');

    expect(buttonElement).toBeInTheDocument();
  });

  it('renders correct text', () => {
    const buttonText = 'Click me';

    const { getByTestId } = render(<Button>{buttonText}</Button>);

    const buttonElement = getByTestId('button');

    expect(buttonElement).toHaveTextContent(buttonText);
  });

  it('calls the onClick function when clicked', () => {
    const onClickMock = vi.fn();

    const { getByTestId } = render(<Button onClick={onClickMock} />);

    const buttonElement = getByTestId('button');

    fireEvent.click(buttonElement);

    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  it('forwards ref', () => {
    const ref = React.createRef<HTMLButtonElement>();

    const { getByTestId } = render(<Button ref={ref} />);

    const buttonElement = getByTestId('button');

    expect(ref.current).toBe(buttonElement);
  });

  describe('snaphots', () => {
    it('variant default', () => {
      const { asFragment } = render(<Button />);
      expect(asFragment()).toMatchSnapshot();
    });

    it('variant desctructive', () => {
      const { asFragment } = render(<Button variant="destructive" />);
      expect(asFragment()).toMatchSnapshot();
    });

    it('variant outline', () => {
      const { asFragment } = render(<Button variant="outline" />);
      expect(asFragment()).toMatchSnapshot();
    });

    it('variant secondary', () => {
      const { asFragment } = render(<Button variant="secondary" />);
      expect(asFragment()).toMatchSnapshot();
    });

    it('variant ghost', () => {
      const { asFragment } = render(<Button variant="ghost" />);
      expect(asFragment()).toMatchSnapshot();
    });

    it('variant link', () => {
      const { asFragment } = render(<Button variant="link" />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
