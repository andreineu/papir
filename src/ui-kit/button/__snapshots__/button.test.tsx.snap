// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ui-button > snaphots > variant default 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-400 disabled:pointer-events-none disabled:opacity-50 dark:focus-visible:ring-neutral-800 bg-neutral-900 text-neutral-50 shadow hover:bg-neutral-900/90 dark:bg-neutral-50 dark:text-neutral-900 dark:hover:bg-neutral-50/90 h-9 px-4 py-2"
    data-testid="button"
  />
</DocumentFragment>
`;

exports[`ui-button > snaphots > variant desctructive 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-400 disabled:pointer-events-none disabled:opacity-50 dark:focus-visible:ring-neutral-800 bg-red-500 text-neutral-50 shadow-sm hover:bg-red-500/90 dark:bg-red-900 dark:text-red-50 dark:hover:bg-red-900/90 h-9 px-4 py-2"
    data-testid="button"
  />
</DocumentFragment>
`;

exports[`ui-button > snaphots > variant ghost 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-400 disabled:pointer-events-none disabled:opacity-50 dark:focus-visible:ring-neutral-800 hover:bg-neutral-100 hover:text-neutral-900 dark:hover:bg-neutral-800 dark:hover:text-neutral-50 h-9 px-4 py-2"
    data-testid="button"
  />
</DocumentFragment>
`;

exports[`ui-button > snaphots > variant link 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-400 disabled:pointer-events-none disabled:opacity-50 dark:focus-visible:ring-neutral-800 text-neutral-900 underline-offset-4 hover:underline dark:text-neutral-50 h-9 px-4 py-2"
    data-testid="button"
  />
</DocumentFragment>
`;

exports[`ui-button > snaphots > variant outline 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-400 disabled:pointer-events-none disabled:opacity-50 dark:focus-visible:ring-neutral-800 border border-neutral-200 bg-white shadow-sm hover:bg-neutral-100 hover:text-neutral-900 dark:border-neutral-800 dark:bg-neutral-950 dark:hover:bg-neutral-800 dark:hover:text-neutral-50 h-9 px-4 py-2"
    data-testid="button"
  />
</DocumentFragment>
`;

exports[`ui-button > snaphots > variant secondary 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-400 disabled:pointer-events-none disabled:opacity-50 dark:focus-visible:ring-neutral-800 bg-neutral-100 text-neutral-900 shadow-sm hover:bg-neutral-100/80 dark:bg-neutral-800 dark:text-neutral-50 dark:hover:bg-neutral-800/80 h-9 px-4 py-2"
    data-testid="button"
  />
</DocumentFragment>
`;
